### PR TITLE
URL Rewriter

### PR DESCRIPTION
**Pull request by jsha**
I put together this prototype tool to rewrite URL references in an existing code base, based on HTTPS Everywhere rules. It runs on node.js - instructions for use are in rewriter.js. Let me know what you think!

### CODE REVIEW

**Comment by cooperq** [comment]
Is this ready for a review?

**Comment by jsha** [comment]
This is pretty old. Lemme un-staleify it first.

**Comment by jsha** [comment]
@cooperq: This is now ready for review, and I'll be basing some upcoming tests on it.

File: chromium/rules.js [hunk 1]
```diff
@@ -337,3 +324,7 @@ RuleSets.prototype = {
     return null;
   }
 };
+
+// Export for HTTPS Rewriter if applicable.
+if (typeof exports != 'undefined')
+  exports.RuleSets = RuleSets;
```
Comment (cooperq) [inline chromium/rules.js]: I really hate bracket-less if statements. But keep it if that's the local style.

File: rewriter/rewriter.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    fileFilter: ['*.html', '*.js', '*.rb', '*.erb', '*.mustache', 
+                 '*.scala', '*.c', '*.cc', '*.cpp', '*.cxx',
+                 '*.java', '*.go', '*.php', '*.css', '*.pl', '*.py',
+                 '*.rhtml', '*.sh', '*.yaml']
+  });
+
+  stream
+  .on('warn', function (err) { 
+    console.error('non-fatal error', err); 
+    // optionally call stream.destroy() here in order to abort and cause 'close' to be emitted
```
Comment (cooperq) [inline rewriter/rewriter.js]: nit: Capitalize O

File: rewriter/rewriter.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+if (process.argv.length <= 2) {
+  usage();
+}
+
+for (var i = 2; i < process.argv.length; i++) {
+  var rewritePath = process.argv[i];
+  if (rewritePath.indexOf('-') == 0)
+    usage();
```
Comment (cooperq) [inline rewriter/rewriter.js]: Another no bracket if statement.

**Comment by cooperq** [comment]
Looks fine other than my nits here. Are we not doing JSDoc comments for functions in HTTPS everywhere?

**Comment by jsha** [comment]
Fixed the things you commented on and added some JSDoc. Please let me know of any functions I missed. I am pro-JSDoc everywhere.

**Comment by cooperq** [comment]
Looks good

**Pull request merged**